### PR TITLE
PLT-639: display full name, nickname or username in direct messages

### PR DESCRIPTION
**Pull request by stasvovk**
Ticket's description says
> b) Change display of user names in direct message list and in direct message channel heading based on this setting.

Do we need to change it in these places only or maybe it makes sense to use this setting everywhere?

### CODE REVIEW

**Comment by it33** [comment]
Thanks @stasvovk!

Um, not totally sure, maybe we can take the change and then see how things look and decide on further changes? 

By the way, the core team now has a nightly builds server, so you can see your contribution after it's accepted, and also discuss tickets in the Tickets channel: https://pre-release.mattermost.com/signup_user_complete/?id=rcgiyftm7jyrxnma1osd8zswby

File: web/react/utils/utils.jsx [hunk 1]
```diff
@@ -884,6 +884,23 @@ export function getDisplayName(user) {
...
+    const user = UserStore.getProfile(userId);
+    const nameFormat = PreferenceStore.getPreference(Constants.Preferences.CATEGORY_DISPLAY_SETTINGS, 'name_format', {value: 'false'}).value;
+
+    let username;
+    if (nameFormat === 'nickname_full_name') {
+        username = user.nickname || getFullName();
+    } else if (nameFormat === 'full_name') {
+        username = getFullName();
+    }
+    if (!username.trim().length) {
```
Comment (hmhealey) [inline web/react/utils/utils.jsx]: This will throw a reference error if nameFormat is username since username will be undefined at that point

**Comment by stasvovk** [comment]
@hmhealey that's fixed

**Comment by hmhealey** [comment]
Looks good to me

**Comment by crspeller** [comment]
+1

**Pull request merged**